go-controller/pkg/cni/cni.go

```diff
@@ -80,79 +78,70 @@ func (pr *PodRequest) cmdAdd() *PodResult {
 		annotation, err = kubecli.GetAnnotationsOnPod(namespace, podName)
 		if err != nil {
 			// TODO: check if err is non recoverable
-			logrus.Warningf("Error while obtaining pod annotations - %v", err)
+			logrus.Warningf("error getting pod annotations: %v", err)
 			return false, nil
 		}
 		if _, ok := annotation["ovn"]; ok {
 			return true, nil
 		}
 		return false, nil
 	}); err != nil {
-		logrus.Errorf("failed to get pod annotation - %v", err)
-		return nil
+		return nil, fmt.Errorf("failed to get pod annotation: %v", err)
 	}
 
 	ovnAnnotation, ok := annotation["ovn"]
 	if !ok {
-		logrus.Errorf("failed to get ovn annotation from pod")
-		return nil
+		return nil, fmt.Errorf("failed to get ovn annotation from pod")
 	}
 
 	podInfo, err := util.UnmarshalPodAnnotation(ovnAnnotation)
 	if err != nil {
-		logrus.Errorf("unmarshal ovn annotation failed: %v", err)
-		return nil
+		return nil, fmt.Errorf("unmarshal ovn annotation failed: %v", err)
 	}
 
 	ingress, egress, err := extractPodBandwidthResources(annotation)
 	if err != nil {
-		logrus.Errorf("failed to parse bandwidth request: %v", err)
-		return nil
+		return nil, fmt.Errorf("failed to parse bandwidth request: %v", err)
 	}
 	podInterfaceInfo := &PodInterfaceInfo{
 		PodAnnotation: *podInfo,
 		MTU:           config.Default.MTU,
 		Ingress:       ingress,
 		Egress:        egress,
 	}
-	podResult := &PodResult{}
 	response := &Response{}
 	if !config.UnprivilegedMode {
 		response.Result = pr.getCNIResult(podInterfaceInfo)
 	} else {
 		response.PodIFInfo = podInterfaceInfo
 	}
-	podResult.Response, _ = json.Marshal(response)
-	return podResult
+	return json.Marshal(response)
 }
 
-func (pr *PodRequest) cmdDel() *PodResult {
-	err := pr.PlatformSpecificCleanup()
-	if err != nil {
-		logrus.Errorf("Teardown error: %v", err)
+func (pr *PodRequest) cmdDel() ([]byte, error) {
+	if err := pr.PlatformSpecificCleanup(); err != nil {
+		return nil, fmt.Errorf("Teardown error: %v", err)
```

Comment:
probably *don't* need to wrap this error though? (It's going to get displayed in a context where "Teardown error" would be redundant.)